### PR TITLE
ports/stm32/boards: Add OLIMEX H407 board support.

### PR DESCRIPTION
**Pull request by SmithChart**
This change adds the OLIMEX H407 support to the STM32 port.
The H407 (https://www.olimex.com/Products/ARM/ST/STM32-H407/) is simliar
to the already existing E407
(https://www.olimex.com/Products/ARM/ST/STM32-E407) but does not support
Ethernet and has a full-size USB-A port instead of a Mini-USB socket.

Both boards use the STM32F407ZGT6 CPU.

This port is basically a copy of the E407 but with changed pinmux:
* Removed Ethernet pin definition
* Removed UART1 (pins are used for other functions)
* Removed UART3 flow control pins (pins are used for other functions)
* Removed SD-Card detect pin (since it is not connected on the H407)

I have added a REPL on UART3 since this is connected to the
U3BOOT-header. This is a 3-pin header with RX, TX and GND that is
inteded for the serial terminal.

Tested:
* Micro-SD Card is detected when inserted on RESET
* REPL on UART3 works

What does not work:
* Did not manage to bring USB device to function.

Signed-off-by: Chris Fiege <cfi@pengutronix.de>

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks for the contribution.

It's a bit confusing that the board's name is H407 yet it uses an F407 MCU... but I guess we can't change the board name.



> Did not manage to bring USB device to function.

If the USB is on the USB HS port then you need config like this:
```
#define MICROPY_HW_USB_HS              (1)
#define MICROPY_HW_USB_HS_IN_FS        (1)
```

**Comment by SmithChart** [comment]
Your changes for USB work. Now the device enumerates fine:

```
[525116.213704] usb 1-2: new full-speed USB device number 63 using xhci_hcd
[525116.367702] usb 1-2: New USB device found, idVendor=f055, idProduct=9800, bcdDevice= 2.00
[525116.367720] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[525116.367729] usb 1-2: Product: Pyboard Virtual Comm Port in FS Mode
[525116.367735] usb 1-2: Manufacturer: MicroPython
[525116.367740] usb 1-2: SerialNumber: 396D376C3231
[525116.371208] usb-storage 1-2:1.0: USB Mass Storage device detected
[525116.374304] scsi host3: usb-storage 1-2:1.0
[525116.375165] cdc_acm 1-2:1.1: ttyACM1: USB ACM device
[525117.382075] scsi 3:0:0:0: Direct-Access     MicroPy  pyboard Flash    1.00 PQ: 0 ANSI: 2
[525117.383252] sd 3:0:0:0: Attached scsi generic sg2 type 0
[525117.383574] sd 3:0:0:0: [sdc] 480 512-byte logical blocks: (246 kB/240 KiB)
[525117.383814] sd 3:0:0:0: [sdc] Write Protect is off
[525117.383822] sd 3:0:0:0: [sdc] Mode Sense: 03 00 00 00
[525117.384004] sd 3:0:0:0: [sdc] No Caching mode page found
[525117.384014] sd 3:0:0:0: [sdc] Assuming drive cache: write through
[525117.417817]  sdc: sdc1
[525117.420506] sd 3:0:0:0: [sdc] Attached SCSI removable disk
```

And I have a serial device and a REPL on the mini USB socket. Thanks!

I have updated my branch. @dpgeorge please take another look :)

File: ports/stm32/boards/OLIMEX_H407/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+CMSIS_MCU = STM32F407xx
+AF_FILE = boards/stm32f405_af.csv
+LD_FILES = boards/stm32f405.ld boards/common_ifs.ld
+TEXT0_ADDR = 0x08000000
+TEXT1_ADDR = 0x08020000
+
+# MicroPython settings
+MICROPY_PY_LWIP = 1
+MICROPY_PY_USSL = 1
+MICROPY_SSL_MBEDTLS = 1
```
Comment (dpgeorge) [inline ports/stm32/boards/OLIMEX_H407/mpconfigboard.mk:11]: since ETH is not enabled, these 3 settings can be removed

Comment (SmithChart) [reply ports/stm32/boards/OLIMEX_H407/mpconfigboard.mk]: Removed these lines and updated my branch.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 6e39f2cc1ed7d4e6a6b1d3132a1b5a73b198ef38

**Pull request closed without merging**